test/common/init/target_impl_test.cc

```diff
@@ -8,15 +8,22 @@ namespace Envoy {
 namespace Init {
 namespace {
 
-TEST(InitTargetImplTest, Name) {
-  ExpectableTargetImpl target;
-  EXPECT_EQ("target test", target.name());
+// Testing common cases for all the target implementation.
+template <typename T> class TargetImplTest : public ::testing::Test {};
+TYPED_TEST_SUITE_P(TargetImplTest);
+
+template <typename T> std::string getName() { return ""; }
```

Comment:
FWIW, you could avoid templates here and do something similar to what was done in https://github.com/envoyproxy/envoy/blob/master/test/common/config/subscription_test_harness.h, but I think this isn't too bad and makes things nice and clean.